lib/rubygems/security/signer.rb

```diff
@@ -40,7 +49,11 @@ def self.re_sign_cert(expired_cert, expired_cert_path, private_key)
 
     Gem::Security.write(expired_cert, new_expired_cert_path)
 
-    re_signed_cert = Gem::Security.re_sign(expired_cert, private_key)
+    re_signed_cert = Gem::Security.re_sign(
+      expired_cert,
+      private_key,
+      (Gem::Security::ONE_DAY * Gem.configuration.cert_expiration_length_days)
```

Comment:
We're calculation the date in quite a few places as well, this might be good to extract into a method somewhere.

Reply:
Both calculations are not exactly the same, so i think it will be better to keep it simple